addon/mixins/select-picker.js

```diff
@@ -44,6 +44,20 @@ var isAdvancedSearch = function(liveSearch) {
   );
 };
 
+var promiseResolver = function(prop) {
+  return Ember.computed(prop, function() {
```

Comment:
I don't this this should be a computed property. Lazy loading would be too late to allow asynchronous resolution. I think this needs to be an observer instead.